crates/cli-support/src/js/mod.rs

```diff
@@ -142,7 +142,7 @@ impl<'a> Context<'a> {
             self.globals.push_str(c);
         }
         let global = match self.config.mode {
-            OutputMode::Node { module: false } => {
+            OutputMode::Isomorphic | OutputMode::Node { module: false } => {
```

Comment:
Why the name `Isomorphic`?

Reply:
It was introduced in https://github.com/rustwasm/wasm-pack/issues/1334 and I didn't question it, it was just a catchy name. The idea seems to be that the code should run unchanged (without polyfiils or other modification) in any context (node, browser, ...). But this goal is beyond what I am doing here. Here I just inline the assets that need (and effectively cannot atm) to be processed by a bundler.
I would rather call it `node-universal`. Basically because that's basically what it should do: You can build your Rust code with `wasm-pack` and include it in your typescript application (e.g. react, aws lambda, node express) without worrying about the bundling.
What do you think?

Reply:
Sorry for disturbing you here.

I previously facilitated the sharing of cjs/esm/bundler for WASM through manual adjustments in this https://github.com/JohnnyMorganz/StyLua/pull/848 .
I also shared my concept in https://github.com/rustwasm/wasm-bindgen/issues/3790, which supports various environments, including browsers, Node, Deno, Bun, and bundlers.

I'm not certain about your reference to `include it in your typescript application (e.g. react, ` but include CJS into the web project is challenging. We faced this issue in https://github.com/swc-project/swc-playground/issues/42, and finallay, we had to create a separate esm to fulfill the requirements.

Inline WASM is viable, but I don't believe it can serve as a direct alternative to the current web/node/deno/bundler option. Instead, this option should complement them, being usable independently and applicable across all environments.

Reply:
> Sorry for disturbing you here.
Thank you for your valuable insights!

> I previously facilitated the sharing the same WebAssembly file across of cjs/esm/bundler through manual adjustments in this [JohnnyMorganz/StyLua#848](https://github.com/JohnnyMorganz/StyLua/pull/848) . I also shared my concept in #3790, which supports various environments, including browsers, Node, Deno, Bun, and bundlers.
> 
> I'm not certain about your reference to `include it in your typescript application (e.g. react, ` but include CJS into the web project is challenging. We faced this issue in [swc-project/swc-playground#42](https://github.com/swc-project/swc-playground/issues/42), and finallay, we had to create a separate esm to fulfill the requirements.

Maybe I am missing something here (I am actually using wasm only in AWS Lambdas for now). Is there an issue with using the npm package generated by the `Isomorphic` target in an arbitrary react app?

> Inline WASM is viable, but I don't believe it can serve as a direct alternative to the current web/node/deno/bundler option. Instead, this option should complement them, being usable independently and applicable across all environments.
I totally agree. My focus is also on the inline part not on the "usability across targets". I have been watching the asset handling issue in `esbuild`, `webpack`, `rollup`, `wasm-pack`, `wasm-bindgen` and never came closer to the solution than writing custom plugins plus a bunch of scripts. So as an intermediate solution I would propose inline wasm until every target just speaks wasm as esm module either directly or with any bundler of my choice without even having to change configuration.

Summing this up, I think I should just focus on an "inline node" target in this PR, moving the portability issues to the future when someone actually needs this.